### PR TITLE
Fix nested Flexible Layout direction problem

### LINKED ISSUE #7635: Nested Flexible Layouts don't display properly
#### Summary
A Flexible Layout (FL) that uses a column layout displays its elements like this:
Screenshot 2024-03-21 at 11 53 29 AM

but, when placed in a FL that uses a rows layout displays incorrectly:
Screenshot 2024-03-21 at 11 54 24 AM

The problem is that outer FL's `flex-direction` is affecting the inner's, but shouldn't.

#### Steps to Reproduce

1. Create a Flexible Layout (FL) using a columns layout. Put two objects inside the same column, and note which object is above the other. Save and exit editing.
2. Create another FL using a rows layout. Drag the DL from step 1 into a row of this layout.
3. Observe: the objects in the step 1 layout will now be side-by-side, but should be above and below each other.

#### Environment

* Open MCT Version:  4.0.0-next
* Deployment Type:  any
* OS: all
* Browser: legion

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by charlesh88**
Closes #7635
- More specific approach to CSS class application for column vs. row layouts.
- Added layout direction CSS classing to `c-fl-container__frames-holder`.
- Switched toolbar icon and titling for better parity with 'toggle' approach used elsewhere.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/flexibleLayout/components/ContainerComponent.vue [hunk 1]
```diff
@@ -45,7 +45,11 @@
       @object-drop-to="moveOrCreateNewFrame"
     />
 
-    <div role="row" class="c-fl-container__frames-holder">
+    <div
+      role="row"
+      class="c-fl-container__frames-holder"
+      :class="[rowsLayout ? '--layout-rows' : '--layout-cols']"
```
Comment (ozyx) [inline src/plugins/flexibleLayout/components/ContainerComponent.vue:51]: Let's move all of this into a computed property

File: src/plugins/flexibleLayout/components/FlexibleLayout.vue [hunk 2]
```diff
@@ -30,10 +30,8 @@
 
     <div
       class="c-fl__container-holder u-style-receiver js-style-receiver"
-      :class="{
-        'c-fl--rows': rowsLayout === true
-      }"
-      :aria-label="`Flexible Layout ${rowsLayout ? 'Row' : 'Column'}`"
+      :class="[rowsLayout ? 'c-fl--rows' : 'c-fl--cols']"
+      :aria-label="`Flexible Layout ${rowsLayout ? 'Rows' : 'Columns'}`"
```
Comment (ozyx) [inline src/plugins/flexibleLayout/components/FlexibleLayout.vue:34]: Let's make these computed properties

File: src/plugins/flexibleLayout/components/flexible-layout.scss [hunk 3]
```diff
@@ -30,23 +29,26 @@
     flex: 0 0 auto;
   }
 
+  &--cols {
+  }
+
+  &--rows {
+  }
```
Comment (ozyx) [inline src/plugins/flexibleLayout/components/flexible-layout.scss:36]: We should remove these empty rulesets if they aren't being used

File: src/plugins/flexibleLayout/toolbarProvider.js [hunk 4]
```diff
@@ -61,13 +61,13 @@ function ToolbarProvider(openmct) {
...
-            title: 'Columns layout'
+            icon: 'icon-rows',
+            title: 'Switch to rows layout'
           },
           {
             value: false,
-            icon: 'icon-rows',
-            title: 'Rows layout'
+            icon: 'icon-columns',
+            title: 'Switch to columns layout'
```
Comment (ozyx) [inline src/plugins/flexibleLayout/toolbarProvider.js:70]: We need to make sure we update any e2e tests that use these titles / labels as locators

**Review by ozyx — CHANGES_REQUESTED** [review]
Looking good so far, have a few comments and suggestions

File: src/plugins/flexibleLayout/toolbarProvider.js (hunk 4 above)
Comment (charlesh88) [reply src/plugins/flexibleLayout/toolbarProvider.js:70]: Updated one test file.

File: src/plugins/flexibleLayout/components/flexible-layout.scss (hunk 3 above)
Comment (charlesh88) [reply src/plugins/flexibleLayout/components/flexible-layout.scss:36]: Good catch, removed.

File: src/plugins/flexibleLayout/components/FlexibleLayout.vue (hunk 2 above)
Comment (charlesh88) [reply src/plugins/flexibleLayout/components/FlexibleLayout.vue:34]: Done

**Comment by charlesh88** [comment]
All change requests addressed, changes pushed including an update to latest master.

**Review by ozyx — CHANGES_REQUESTED** [review]
Good stuff! Couple things remaining:

- need to fix one failing test which looks to be due to changes in flexible layout behavior
- this PR needs a quick regression test

**Review by ozyx — APPROVED** [review]

**Pull request merged**